### PR TITLE
Add tests and fix LD_PRELOAD propagation

### PR DESCRIPTION
**Pull request by Mrokkk**

### CODE REVIEW

File: bin/yebash [hunk 1]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-LD_PRELOAD=$DIR/../lib/libyebash.so /bin/bash
+LD_PRELOAD=$DIR/../lib/libyebash.so /bin/bash --rcfile <(echo 'unset LD_PRELOAD; . ~/.bashrc')
```
Comment (szborows) [inline bin/yebash]: a question: will `~` properly evaluate in all environments?

File: test/helpers/Helpers.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "../../src/History.hpp"
```
Comment (szborows) [inline test/helpers/Helpers.hpp]: shouldn't we use `include_directories` in CMake instead?

File: test/HistorySuggestion.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include "../src/HistorySuggestion.hpp"
```
Comment (szborows) [inline test/HistorySuggestion.cpp]: :/

File: test/HistorySuggestion.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#include "helpers/Helpers.hpp"
+#include "catch.hpp"
+
+using namespace yb;
+
+namespace {
+
+void testSuggestions(HistorySuggestion &suggestion, std::string chars, std::initializer_list<std::string> expected) {
+    for (std::pair<std::string::const_iterator, std::initializer_list<std::string>::const_iterator> it{chars.begin(), expected.begin()};
```
Comment (szborows) [inline test/HistorySuggestion.cpp]: isn't `auto` working?

File: test/History.cpp [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+#include <initializer_list>
+
+using namespace yb;
+
+TEST_CASE( "can read commands from input", "[History.read]" ) {
+    std::initializer_list<std::string> inputList{"", "test", "test2", "test3", "abcd", "a", "dcba", "", "test4", ""};
+    auto history = Helpers::createHistory(inputList);
+    size_t i = 0;
+    for (std::pair<std::initializer_list<std::string>::iterator, History::const_iterator> it(inputList.end() - 1, history.begin());
```
Comment (szborows) [inline test/History.cpp]: `auto`?

File: bin/yebash (hunk 1 above)
Comment (Mrokkk) [inline bin/yebash]: Accordng to the POSIX - yes (http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html chapter 2.6.1)

File: test/helpers/Helpers.hpp (hunk 2 above)
Comment (Mrokkk) [inline test/helpers/Helpers.hpp]: OK, I would fix it

File: test/HistorySuggestion.cpp (hunk 4 above)
Comment (Mrokkk) [inline test/HistorySuggestion.cpp]: I iterate through 2 containers, and `auto` gives me this: `error: initializer for variable 'it' with type 'auto' contains multiple expressions`

File: test/History.cpp (hunk 5 above)
Comment (Mrokkk) [inline test/History.cpp]: As above ;)

File: bin/yebash (hunk 1 above)
Comment (szborows) [inline bin/yebash]: nice. thanks for providing a link!

File: test/HistorySuggestion.cpp (hunk 4 above)
Comment (szborows) [inline test/HistorySuggestion.cpp]: then, maybe make use of `std::make_pair`?

**Pull request merged**